test/ngResource/resourceSpec.js

```diff
@@ -97,21 +97,50 @@ describe('basic usage', function() {
     $httpBackend.flush();
   });
 
-  it('should include a request body when calling custom delete with hasBody is true', function() {
+  it('should include a request body when calling custom method with hasBody is true', function() {
+    var instant = {name: 'info.txt', value: 'V2hlbiB0aGUgdGltZSBlbmRzLg=='};
+    var fid = 42;
+    var created = {fid: fid, filname: 'fooresource', value: 'V2hlbiB0aGUgdGltZSBlbmRzLg=='};
     var condition = {at: '2038-01-19 03:14:08'};
+    $httpBackend.expect('CREATE', '/fooresource', instant).respond(created);
     $httpBackend.expect('DELETE', '/fooresource', condition).respond({});
 
     var r = $resource('/fooresource', {}, {
+      create: {method: 'CREATE', hasBody: true},
       delete: {method: 'DELETE', hasBody: true}
     });
 
+    var creationResponse = r.create(instant);
     var deleteResponse = r.delete(condition);
 
     $httpBackend.flush();
 
+    expect(creationResponse.fid).toBe(fid);
     expect(deleteResponse.$resolved).toBe(true);
   });
 
+  it('should not include a request body if hasBody is false on POST, PUT and PATCH', function() {
+    $httpBackend.expect('POST', '/foo').respond({});
+    $httpBackend.expect('PUT', '/foo').respond({});
+    $httpBackend.expect('PATCH', '/foo').respond({});
+
+    var R = $resource('/foo', {}, {
+      post: {method: 'POST', hasBody: false},
+      put: {method: 'PUT', hasBody: false},
+      patch: {method: 'PATCH', hasBody: false}
+    });
+
+    var postResponse = R.post();
+    var putResponse = R.put();
+    var patchResponse = R.patch();
```

Comment:
If you pass two params (e.g. `R.post(null, {foo: 'bar'})`), then the first one is interpreted as `params` and the second one as `data`.

For example, th following test currently fails, because `{id: -1}` is not undefined. If something similar passes with your PR, then we are good:

```js
    $httpBackend.expectPOST('/foo').respond(function(method, url, data) {
      expect(data).not.toBeUndefined();
      return [200, {id: 42}];
    });

    var R = $resource('/foo');
    var response = R.save(null, {id: -1});

    $httpBackend.flush();
    expect(response.id).toBe(42);
```